cmd/image-builder/main.go

```diff
@@ -370,6 +369,23 @@ func cmdDescribeImg(cmd *cobra.Command, args []string) error {
 	return describeImage(res, osStdout)
 }
 
+var osGetuid = os.Getuid
+
+func imageBuilderDefaultCacheDir() (string, error) {
+	if osGetuid() == 0 {
+		return "/var/cache/image-builder/store", nil
+	}
+	xdgCacheHome := os.Getenv("XDG_CACHE_HOME")
+	if xdgCacheHome == "" {
+		homeDir, err := os.UserHomeDir()
+		if err != nil {
+			return "", err
+		}
+		xdgCacheHome = filepath.Join(homeDir, ".cache")
+	}
+	return filepath.Join(xdgCacheHome, "image-builder/store"), nil
+}
+
```

Comment:
Can we split this into a separate PR? I like it but I think we'll have some more discussion in this PR.

Reply:
Sure - I'm curious about your thoughts here, what extra discussion do you have in min? This change  in itself is only useful if in combination with supermin, otherwise there will never be a user cache with actual files (as only manifest can run as non-root without supermin)

Reply:
Only disk image types require privileges, there are a few that don't.

Reply:
That is interesting. We do require bind mount in osbuild (and maybe more for bwrap) so a user container with --privileged might be enough for non-disk types (like contianer). I have not tested this though.